utils/node_local_test.py

```diff
@@ -1206,6 +1204,25 @@ def setup_log_test(conf):
 
     lt.wf = conf.wf
 
+def compress_file(filename):
+    """Compress a file using bz2 for space reasons"""
+    small = bz2.BZ2Compressor()
+
+    fd = open(filename, 'rb')
+
+    nfd = open('{}.bz2'.format(filename), 'wb')
+    lines = fd.read(64*1024)
+    while lines:
+        new_data = bz2.compress(lines)
+        if new_data:
+            nfd.write(new_data)
+        lines = fd.read(64*1024)
+    new_data = small.flush()
+    if new_data:
+        nfd.write(new_data)
+
+    os.unlink(filename)
```

Comment:
will os.unlink work without doing explicit fd.close() on it first?

Reply:
Yes.  The fd will be closed when it goes out of scope, so when this function returns.

Reply:
Just curious, but why not use the file IO methods in https://github.com/python/cpython/blob/3.9/Lib/bz2.py?

Reply:
In case of very large files.  CI doesn't generate them, but NLT certainly can if you want it to.

Reply:
Are you saying the IO methods in the library will have issues with very large files?

Reply:
I'm not sure what you're getting at, the compression function in that link takes a buffer, so will fail if the file size is large enough yes.